### DISCUSSION TITLE
Relume

### DISCUSSION THREAD

**Discussion by kof** [webstudio-is/webstudio-community#12] [Ideas]
This Webflow components library sounds awesome. We should see how we can support importing those components as a user easily.

https://twitter.com/relume_io

https://library.relume.io/

**Comment by ghost** [comment]
I'mnot sure how we would be able to use Webflow components in Webstudio but besides that...

Relume gets a lot of chatter on Twitter but I've tried it and it's not very useful.

It's pre-built and pre-styled components. Maybe helpful to someone building without a design in mind, but I never have and never will start building without a design. When there's a custom design to be followed, Relume just gets in the way.

It takes even more time to re-name all the classes and make the necessary style adjustments than it does to just start from scratch. If one knows how to build these things from scratch, then they can do that faster than importing a new Relume component and adjusting it to fit the design. If they don't know how to build these components, then it doesn't help them to borrow from Relume because they won't understand how to customize it or troubleshoot.

So for me, Relume is completely useless.

However in the distant future I do think it would be great to have our own component library that you can access directly from within the program. The implementation of this depends on how we do tokens & components. Webflow has a bare bones version of this but it's not much, which is why 3rd party solutions like Relume have been showing up.

**Reply by kof** [reply]
Good feedback! Let's keep an eye on Relume and learn who is using them and how. My guess its for people who design in webflow directly and who don't do much custom stuff, but basically put together the Relume components. So they basically compose the site they need out of relume and then add styles to those classes without renaming them. If that workflow makes sense to a lot of people - we should think how to do it.


I think we should have generally a way to import or copy/paste webflow stuff, so that would make relume work automatically. But also we need to see in detail how bad this works when you have tokens and components and if that would make it not a great component in the end.

What I really want is an official way for us to have a marketplace for components, which would be not just relume but hundreds of other components created by different people. Relume could be a great source of inspiration for that, it just shows how many building blocks there can be.

Once Webstudio is a better "platform", maybe Relume guys will build a library for Webstudio 😅

**Reply by ghost** [reply]
Let's look into how Framer does components.

I once saw Framer support on Twitter giving someone a component to use in their project by saying "paste this URL in your project". 

If it could be that simple that would be a dream

**Reply by smotched** [reply]
Can go even further, an extension/plugin system in which components are just one part. Where people can open source plugins that integrate/extend webstudio in some ways.

**Reply by kof** [reply]
yeah, for example similar to how CSF (https://github.com/webstudio-is/webstudio-designer/issues/24) lets user (SDK) define argTypes (controls) for properties, we are thinking to also let the SDK define the control component that is rendered in the style panel. So a component author can change how their component is being controlled from style panel.

This was just an example, no need to go deep into it in this scope, but yes we will look into enabling user bringing their behavior/customization beyond components on the canvas. 

For now, we need to build the foundation first and once we are there anything that allows more people use it more efficiently (growth) should be a priority. That's why I brought up Relume.

**Reply by Ren2049** [reply]
@taylornowotny 

> If it could be that simple that would be a dream

WordPress block patterns/components work like that as well (probably some good initial open source inspiration there). It's hands down the fastest way to build, especially for self-servicing end users because they don't even have to deal with the detailed controls in the editor (the one in WP has some UX issues).

As a web agency you build out your pattern library for a niche and/or customize it for a particular client and then instruct the clients to drop-in the patterns they need.

Imo libraries like Relume (or pattern libraries bundled with generic WP themes) provide a good starting point for generic brochure sites. I think component generating Ai is really interesting. Not a fan of the Framer Ai though, Ai can't make good design decisions yet. Deciding and curating the results on a per component basis is likely the way to go.
